crates/wasi-provider/src/lib.rs

```diff
@@ -85,185 +82,63 @@ impl WasiProvider {
-        &self,
-        container: &Container,
-        pod: &Pod,
-        client: &kube::client::Client,
-        modules: &mut HashMap<String, Vec<u8>>,
-        volumes: &HashMap<String, Ref>,
-    ) -> anyhow::Result<
-        kubelet::container::Handle<wasi_runtime::Runtime, wasi_runtime::HandleFactory>,
-    > {
-        let env = Self::env_vars(&container, pod, &client).await;
-        let args = container.args().clone().unwrap_or_default();
-        let module_data = modules
-            .remove(container.name())
-            .expect("FATAL ERROR: module map not properly populated");
-        let container_volumes = Self::volume_path_map(container, volumes)?;
-
-        let runtime = WasiRuntime::new(
-            module_data,
-            env,
-            args,
-            container_volumes,
-            self.log_path.clone(),
-        )
-        .await?;
-
-        debug!("Starting container {} on thread", container.name());
-        let handle = runtime.start().await?;
-
-        Ok(handle)
-    }
-
-    async fn start_app_containers(
-        &self,
-        pod: &Pod,
-        client: &kube::client::Client,
-        modules: &mut HashMap<String, Vec<u8>>,
-        volumes: &HashMap<String, Ref>,
-    ) -> (
-        ContainerHandleMap<wasi_runtime::Runtime, wasi_runtime::HandleFactory>,
-        Option<anyhow::Error>,
-    ) {
-        info!("Starting containers for pod {:?}", pod.name());
-        let mut container_handles = HashMap::new();
-        for container in pod.containers() {
-            let start_result = self
-                .start_container(&container, pod, client, modules, volumes)
-                .await;
-            match start_result {
-                Ok(handle) => {
-                    container_handles
-                        .insert(ContainerKey::App(container.name().to_owned()), handle);
-                }
-                Err(e) => {
-                    return (container_handles, Some(e));
-                }
-            }
-        }
-        info!(
-            "All containers started for pod {:?}. Updating status",
-            pod.name()
-        );
-
-        (container_handles, None)
-    }
+struct ModuleRunContext {
+    modules: HashMap<String, Vec<u8>>,
+    volumes: HashMap<String, Ref>,
+}
 
-    async fn run_container_to_completion(
-        &self,
-        status_receiver: &mut tokio::sync::watch::Receiver<kubelet::container::Status>,
-        container: &Container,
-    ) -> anyhow::Result<()> {
-        let result = Self::wait_for_terminated_status(status_receiver).await;
-        debug!("Init container {} terminated", container.name());
-        if result.succeeded {
-            Ok(())
-        } else {
-            Err(anyhow::anyhow!(
-                "Init container {} failed with message {}",
-                container.name(),
-                result.message
-            ))
-        }
-    }
+fn make_status(phase: Phase, reason: &str) -> anyhow::Result<serde_json::Value> {
```

Comment:
Should we move this function to `kubelet` for now?

Reply:
I agree.